
<img src="Tp1.png">


### Importer les bibliothèques **numpy** et **pandas** et importer functools de reduce

In [2]:
# Importer les bibliothèques nécessaires




In [3]:
import pandas as pd
import numpy as np
from functools import reduce 

# PART I

Explorer la dataset `BL-Flickr-Images-Book.csv` 

**NB:** faites attentionsau chemin

## 1 Exploration du dataset

In [4]:
# Exporter BL-Flickr-Images-Book.csv dans un data frame nomé df
df = pd.read_csv('BL-Flickr-Images-Book.csv')

# Afficher les 5 premières lignes
df.head(25)

# Afficher les neme premières lignes : df.head(n)




Identifier                                  Edition Statement  \
0          206                                                NaN   
1          216                                                NaN   
2          218                                                NaN   
3          472                                                NaN   
4          480                       A new edition, revised, etc.   
5          481                      Fourth edition, revised, etc.   
6          519                                                NaN   
7          667                                                NaN   
8          874                                                NaN   
9         1143                                                NaN   
10        1280                                                NaN   
11        1808                                                NaN   
12        1905                                                NaN   
13        1929                                                NaN   
14        2836                                                NaN   
15        2854                                                NaN   
16        2956                                                NaN   
17        2957                                                NaN   
18        3017  Nueva edicion, anotada ... y continuada ... po...   
19        3131                                                NaN   
20        4598                                                NaN   
21        4884                                                NaN   
22        4976  [Another edition.] Abdollatiphi historiæ Ægypt...   
23        5382                                                NaN   
24        5385      [Another edition.] Illustrated by John Leech.   

                   Place of Publication Date of Publication  \
0                                London         1879 [1878]   
1              London; Virtue & Yorston                1868   
2                                London                1869   
3                                London                1851   
4                                London                1857   
5                                London                1875   
6                                London                1872   
7   pp. 40. G. Bryan & Co: Oxford, 1898                 NaN   
8                               London]                1676   
9                                London                1679   
10                             Coventry                1802   
11                          Christiania                1859   
12                              Firenze                1888   
13                            Amsterdam         1839, 38-54   
14                               Savona                1897   
15                               London                1865   
16                                Paris             1860-63   
17                                Paris                1873   
18                          Puerto-Rico                1866   
19                             New York                1899   
20                                 Hull                1814   
21                               London                1820   
22                               Oxonii                1800   
23                               London  1847, 48 [1846-48]   
24                               London             [1897?]   

                Publisher                                              Title  \
0        S. Tinsley & Co.                  Walter Forbes. [A novel.] By A. A   
1            Virtue & Co.  All for Greed. [A novel. The dedication signed...   
2   Bradbury, Evans & Co.  Love the Avenger. By the author of “All for Gr...   
3           James Darling  Welsh Sketches, chiefly ecclesiastical, to the...   
4    Wertheim & Macintosh  [The World in which I live, and my place in it...   
5       William Macintosh  [The World in which I live, and my place in it...   
6         

## 2 Jeter un coup d'oeil dans le dataset

In [5]:
# des informations sur DATASET
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8287 entries, 0 to 8286
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Identifier              8287 non-null   int64  
 1   Edition Statement       773 non-null    object 
 2   Place of Publication    8287 non-null   object 
 3   Date of Publication     8106 non-null   object 
 4   Publisher               4092 non-null   object 
 5   Title                   8287 non-null   object 
 6   Author                  6509 non-null   object 
 7   Contributors            8287 non-null   object 
 8   Corporate Author        0 non-null      float64
 9   Corporate Contributors  0 non-null      float64
 10  Former owner            1 non-null      object 
 11  Engraver                0 non-null      float64
 12  Issuance type           8287 non-null   object 
 13  Flickr URL              8287 non-null   object 
 14  Shelfmarks              8287 non-null   

## 3 Qu'est ce que vous remarquez?

In [6]:
# Rédiger votre réponse




## 4 Dropping unnecessary columns

Nous allons restreindre l'étude à seulement 7 colomnes pour cela nous allons supprimer 7 colonnes que nous jugeons unitiles por le moment :
```Python
to_drop = ['Edition Statement',
           'Corporate Author',
           'Corporate Contributors',
           'Former owner',
           'Engraver',
           'Contributors',
           'Issuance type',
           'Shelfmarks']```

In [7]:
to_drop = ['Edition Statement',
           'Corporate Author',
           'Corporate Contributors',
           'Former owner',
           'Engraver',
           'Contributors',
           'Issuance type',
           'Shelfmarks']

#Eliminer les colonnes dans la liste de to_drop du dataframe df
#les paremetre : inplace : pour faire la modification sur le data originalet axis :pour dit supprime des ligne si axis = 1 donc on supprime des ligne 
df.drop(to_drop, inplace = True , axis =1)


#Afficherle résultat


## 5 Setting the index of the dataset
Changer l'index pourqu'ille prenne la valeur de l'identifiant `Identifier`

In [8]:
###Changerl'index 
# supprime le Identifier qui ajoute par defaut a cause de df.info()
df.set_index('Identifier', inplace =True)

###Afficher le résultat
df.head(50)

Place of Publication Date of Publication  \
Identifier                                                            
206                                      London         1879 [1878]   
216                    London; Virtue & Yorston                1868   
218                                      London                1869   
472                                      London                1851   
480                                      London                1857   
481                                      London                1875   
519                                      London                1872   
667         pp. 40. G. Bryan & Co: Oxford, 1898                 NaN   
874                                     London]                1676   
1143                                     London                1679   
1280                                   Coventry                1802   
1808                                Christiania                1859   
1905                                    Firenze                1888   
1929                                  Amsterdam         1839, 38-54   
2836                                     Savona                1897   
2854                                     London                1865   
2956                                      Paris             1860-63   
2957                                      Paris                1873   
3017                                Puerto-Rico                1866   
3131                                   New York                1899   
4598                                       Hull                1814   
4884                                     London                1820   
4976                                     Oxonii                1800   
5382                                     London  1847, 48 [1846-48]   
5385                                     London             [1897?]   
5389                                     London             [1897?]   
5432                                     Milano                1893   
6036                                     London                1805   
6821                                   Aberdeen                1837   
7521                                       Wien                1896   
7630                                   Abingdon                1898   
8239                                     Quebec                1899   
8435                                     London                1892   
8440                                     London                1894   
11361                                   Leipzig            [1894-96   
11852        pp. 40. W. Cann: Plymouth, [1876?]                 NaN   
13074                                    London                1899   
13364                                    London              [1885]   
14466                                    London             [1860?]   
14703                                      1845                1846   
15141                                    London                1893   
15146                                    London                1894   
16173                                       enk                1817   
16543                                    London                1816   
16544                                    Boston                1817   
17602                                    London                1897   
17752                                   Upsaliæ              [1833]   
18173                                    London                1804   
18218                                 Edinburgh                1859   
18347                                   Yverdon                1777   

                                                    Publisher  \
Identifier                                                      
206                                          S. Tinsley & Co.   
216                                              Virtue & Co.   
218                                     Bradbury, Evans & Co.   
472      

## 6 Donner un appérçu sur la colonne `Date of Publication` 


In [9]:
### Afficher  les 25 premières valeurs de Date of Publication
df['Date of Publication'].head(25)


Identifier
206            1879 [1878]
216                   1868
218                   1869
472                   1851
480                   1857
481                   1875
519                   1872
667                    NaN
874                   1676
1143                  1679
1280                  1802
1808                  1859
1905                  1888
1929           1839, 38-54
2836                  1897
2854                  1865
2956               1860-63
2957                  1873
3017                  1866
3131                  1899
4598                  1814
4884                  1820
4976                  1800
5382    1847, 48 [1846-48]
5385               [1897?]
Name: Date of Publication, dtype: object

> Remarque : Observer la date de publication qui correspond à l'index 206.
> Ca nécessite un nettoyage et une correction : il faut éliminer certains symboles 

## 7 Cleaning columns using the `.apply` function

### Ecrire une fonction qui permet de nettoyer la colonne des dates 

In [10]:
unwanted_characters = ['[', ',', '-']
# Completer la fonction suivante
def clean_dates(item):
    dop= str(item.loc['Date of Publication'])
    
    if dop == 'nan' or dop[0] == '[':
        pass
    
    for character in unwanted_characters:
        if character in dop:
            pass
            pass
    
    return dop

df['Date of Publication'] = df.apply(clean_dates, axis = 1)

In [11]:
df.head()

Place of Publication Date of Publication  \
Identifier                                                 
206                           London         1879 [1878]   
216         London; Virtue & Yorston                1868   
218                           London                1869   
472                           London                1851   
480                           London                1857   

                        Publisher  \
Identifier                          
206              S. Tinsley & Co.   
216                  Virtue & Co.   
218         Bradbury, Evans & Co.   
472                 James Darling   
480          Wertheim & Macintosh   

                                                        Title     Author  \
Identifier                                                                 
206                         Walter Forbes. [A novel.] By A. A      A. A.   
216         All for Greed. [A novel. The dedication signed...  A., A. A.   
218         Love the Avenger. By the author of “All for Gr...  A., A. A.   
472         Welsh Sketches, chiefly ecclesiastical, to the...  A., E. S.   
480         [The World in which I live, and my place in it...  A., E. S.   

                                                   Flickr URL  
Identifier                                                     
206         http://www.flickr.com/photos/britishlibrary/ta...  
216         http://www.flickr.com/photos/britishlibrary/ta...  
218         http://www.flickr.com/photos/britishlibrary/ta...  
472         http://www.flickr.com/photos/britishlibrary/ta...  
480         http://www.flickr.com/photos/britishlibrary/ta...

In [12]:
#Voici une autre méthode
#exécuter pour voir le résultat
unwanted_characters = ['[', ',', '-']

def clean_dates(dop):
    dop = str(dop)
    if dop.startswith('[') or dop == 'nan':
        return 'NaN'
    for character in unwanted_characters:
        if character in dop:
            character_index = dop.find(character)
            dop = dop[:character_index] #iraje3 awel parte 9bal position mta3 caractere ilman7obohimch
    return dop

df['Date of Publication'] = df['Date of Publication'].apply(clean_dates)
df.head(10)
df['Date of Publication'].head(25)

Identifier
206     1879 
216      1868
218      1869
472      1851
480      1857
481      1875
519      1872
667       NaN
874      1676
1143     1679
1280     1802
1808     1859
1905     1888
1929     1839
2836     1897
2854     1865
2956     1860
2957     1873
3017     1866
3131     1899
4598     1814
4884     1820
4976     1800
5382     1847
5385      NaN
Name: Date of Publication, dtype: object

## 8 Observer la collone `Author`
### Completer la fonction `clean_author_names`qui permet de corriger les valeurs de cet colonne

In [13]:
# compléter la fonction suivante ( à la place de pass)
#cree un fonction qui permete de correction la attribut author
def clean_author_names(author):
    
    author = str(author)
    
    if author == 'nan':
        return 'NaN'
    
    author = author.split(',')

    if len(author) == 1:
        name = filter(lambda x: x.isalpha(), author[0])
        return reduce(lambda x, y: x + y, name)
    
    last_name, first_name = author[0], author[1]

    first_name = first_name[:first_name.find('-')] if '-' in first_name else first_name
    
    if first_name.endswith(('.', '.|')):
        parts = first_name.split('.')
        
        if len(parts) > 1:
            first_occurence = first_name.find('.')
            final_occurence = first_name.find('.',first_occurence + 1)
            print(final_occurence)
            first_name = first_name[:first_name.find('.')]
        else:
            first_name = first_name[:first_name.find('')]
    
    last_name = last_name.capitalize()
    
    return f'{first_name} {last_name}'


df['Author'] = df['Author'].apply(clean_author_names)

5
5
5
5
5
5
5
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
8
5
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
5
5
-1
5
5
5
5
5
5
5
5
-1
5
-1
5
5
-1
-1
5
5
5
5
5
5
-1
-1
5
5
5
-1
-1
-1
-1
-1
-1
-1
5
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
5
-1
-1
-1
-1
-1
-1
-1
5
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
5
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
5
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
5
15
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1

# Résultat attendu 
<img src="Datasets\capture42.png">

In [14]:
df.head()

Place of Publication Date of Publication  \
Identifier                                                 
206                           London               1879    
216         London; Virtue & Yorston                1868   
218                           London                1869   
472                           London                1851   
480                           London                1857   

                        Publisher  \
Identifier                          
206              S. Tinsley & Co.   
216                  Virtue & Co.   
218         Bradbury, Evans & Co.   
472                 James Darling   
480          Wertheim & Macintosh   

                                                        Title Author  \
Identifier                                                             
206                         Walter Forbes. [A novel.] By A. A     AA   
216         All for Greed. [A novel. The dedication signed...   A A.   
218         Love the Avenger. By the author of “All for Gr...   A A.   
472         Welsh Sketches, chiefly ecclesiastical, to the...   E A.   
480         [The World in which I live, and my place in it...   E A.   

                                                   Flickr URL  
Identifier                                                     
206         http://www.flickr.com/photos/britishlibrary/ta...  
216         http://www.flickr.com/photos/britishlibrary/ta...  
218         http://www.flickr.com/photos/britishlibrary/ta...  
472         http://www.flickr.com/photos/britishlibrary/ta...  
480         http://www.flickr.com/photos/britishlibrary/ta...

##  Observer la colone `title`
### 9 Completer la fonction `clean_title`qui permet de corriger les valeurs de cet colonne

# Résultat attendu 
 <img src="Datasets/capture43.png">

In [15]:
## complétrer la fonction suivante ( à la place de pass)

def clean_title(title):
    
    if title == 'nan':
        return 'NaN'
    
    if title[0] == '[':
        title = title[1:title.find(']')]
        
    if 'by' in title:
        title = title[:title.find('by')]
    elif 'By' in title:
        title = title[:title.find('by')]

        
    if '[' in title:
        title = title[:title.find('[')]

    title = title[:-2]
        
    title = list(map(str.capitalize, title.split()))
    return ' '.join(title)
    
df['Title'] = df['Title'].apply(clean_title)
df.head()

Place of Publication Date of Publication  \
Identifier                                                 
206                           London               1879    
216         London; Virtue & Yorston                1868   
218                           London                1869   
472                           London                1851   
480                           London                1857   

                        Publisher  \
Identifier                          
206              S. Tinsley & Co.   
216                  Virtue & Co.   
218         Bradbury, Evans & Co.   
472                 James Darling   
480          Wertheim & Macintosh   

                                                        Title Author  \
Identifier                                                             
206                                             Walter Forbes     AA   
216                                             All For Greed   A A.   
218         Love The Avenger. By The Author Of “all For Gr...   A A.   
472         Welsh Sketches, Chiefly Ecclesiastical, To The...   E A.   
480         The World In Which I Live, And My Place In It....   E A.   

                                                   Flickr URL  
Identifier                                                     
206         http://www.flickr.com/photos/britishlibrary/ta...  
216         http://www.flickr.com/photos/britishlibrary/ta...  
218         http://www.flickr.com/photos/britishlibrary/ta...  
472         http://www.flickr.com/photos/britishlibrary/ta...  
480         http://www.flickr.com/photos/britishlibrary/ta...

# PART II
## 1 Cleaning entire dataset
### On considère le fichier suivant 'university_towns.txt'

In [17]:
!head Datasets/university_towns.txt

'head' n’est pas reconnu en tant que commande interne
ou externe, un programme exécutable ou un fichier de commandes.


In [19]:
with open('C:/Users/dell/Desktop/Cours 3DNI/DATAmining/university_towns.txt', 'r') as text:
    textfile = text.read()
    print(textfile)

Alabama[edit]
Auburn (Auburn University)[1]
Florence (University of North Alabama)
Jacksonville (Jacksonville State University)[2]
Livingston (University of West Alabama)[2]
Montevallo (University of Montevallo)[2]
Troy (Troy University)[2]
Tuscaloosa (University of Alabama, Stillman College, Shelton State)[3][4]
Tuskegee (Tuskegee University)[5]
Alaska[edit]
Fairbanks (University of Alaska Fairbanks)[2]
Arizona[edit]
Flagstaff (Northern Arizona University)[6]
Tempe (Arizona State University)
Tucson (University of Arizona)
Arkansas[edit]
Arkadelphia (Henderson State University, Ouachita Baptist University)[2]
Conway (Central Baptist College, Hendrix College, University of Central Arkansas)[2]
Fayetteville (University of Arkansas)[7]
Jonesboro (Arkansas State University)[8]
Magnolia (Southern Arkansas University)[2]
Monticello (University of Arkansas at Monticello)[2]
Russellville (Arkansas Tech University)[2]
Searcy (Harding University)[5]
California[edit]
Angwin (Pacific Union College

## 2 Créer un pandas data frame à partir d'un fichier txt
On veut créer un DataFrame nommé `towns_df` à partir du fichier 'university_towns.txt'

Nous voyons que nous avons des noms d'états périodiques suivis des villes universitaires de cet état: StateA TownA1 TownA2 StateB TownB1 TownB2 .... Si nous regardons la façon dont les noms d'états sont écrits dans le fichier, nous verrons qu'ils ont tous la sous-chaîne «[edit]» en eux.

Nous pouvons tirer parti de ce modèle en créant une liste de tuples (état, ville) et en enveloppant cette liste dans un DataFrame:


In [23]:
#lire le fichier ligne par ligne si on trouve [edit] n7toha state w ilba3edha ilkol line  jusqua trouve [edit] mara o5ra
university_towns = []
with open('C:/Users/dell/Desktop/Cours 3DNI/DATAmining/university_towns.txt') as file:
    for line in file:
        if '[edit]' in line:
             # se rappeler de ce `state` jusqu'à ce qu'on trouve le suivant
            current_state = line.strip()
        else:
             # Sinon, nous avons une ville; garder `state` 
            university_towns.append((current_state, line.strip()))

university_towns[:5]

[('Alabama[edit]', 'Auburn (Auburn University)[1]'),
 ('Alabama[edit]', 'Florence (University of North Alabama)'),
 ('Alabama[edit]', 'Jacksonville (Jacksonville State University)[2]'),
 ('Alabama[edit]', 'Livingston (University of West Alabama)[2]'),
 ('Alabama[edit]', 'Montevallo (University of Montevallo)[2]')]

In [25]:
towns_df = pd.DataFrame(university_towns,
                        columns=['State', 'RegionName'])
towns_df.head()

State                                       RegionName
0  Alabama[edit]                    Auburn (Auburn University)[1]
1  Alabama[edit]           Florence (University of North Alabama)
2  Alabama[edit]  Jacksonville (Jacksonville State University)[2]
3  Alabama[edit]       Livingston (University of West Alabama)[2]
4  Alabama[edit]         Montevallo (University of Montevallo)[2]

## 3 Compléter la fonction `clean_up` pour nettoyer les colonnes de `towns_df`

In [26]:
## complétrer la fonction suivante ( à la place de pass)
def clean_up(item):
    if '(' in item:
        pass
        return item[:item.find('(') - 1]
    
    if '[' in item:
        return item[:item.find('[')]
    

towns_df =  towns_df.applymap(clean_up)
towns_df

State     RegionName
0      Alabama         Auburn
1      Alabama       Florence
2      Alabama   Jacksonville
3      Alabama     Livingston
4      Alabama     Montevallo
..         ...            ...
512  Wisconsin    River Falls
513  Wisconsin  Stevens Point
514  Wisconsin       Waukesha
515  Wisconsin     Whitewater
516    Wyoming        Laramie

[517 rows x 2 columns]

# PART III ( OPTIONNELLE) 
## 1 Renaming columns and skipping rows
### 1.1 Convertir le fichier `olympics.csv` en dataframe pandas `olympics_df`

In [30]:

# Charger le fichier CSV en un DataFrame
olympics_df = pd.read_csv('olympics.csv')

# Afficher les premières lignes du DataFrame pour vérification
print(olympics_df.to_string(index=False))

                                           0        1    2    3    4     5        6    7    8    9    10      11   12   13   14             15
                                         NaN ? Summer 01 ! 02 ! 03 ! Total ? Winter 01 ! 02 ! 03 ! Total ? Games 01 ! 02 ! 03 ! Combined total
                           Afghanistan (AFG)       13    0    0    2     2        0    0    0    0     0      13    0    0    2              2
                               Algeria (ALG)       12    5    2    8    15        3    0    0    0     0      15    5    2    8             15
                             Argentina (ARG)       23   18   24   28    70       18    0    0    0     0      41   18   24   28             70
                               Armenia (ARM)        5    1    2    9    12        6    0    0    0     0      11    1    2    9             12
                     Australasia (ANZ) [ANZ]        2    3    4    5    12        0    0    0    0     0       2    3    4    5             12

### 1.2 Supprimer la première ligne 

In [31]:
# exporter 'olympics.csv' sous forme de data frame nommé olympics_df en supprimant la première ligne
# pour choisier le ligne de header avec le mot header 
#skiprows = 1 pour passe la ligne 1 et commencer data de tableau apres ligne 1
olympics_df = pd.read_csv('C:/Users/dell/Desktop/Cours 3DNI/DATAmining/olympics.csv', skiprows = 1, header = 0)

# Afficher les 5 premières lignes
olympics_df.head()

Unnamed: 0  ? Summer  01 !  02 !  03 !  Total  ? Winter  \
0        Afghanistan (AFG)        13     0     0     2      2         0   
1            Algeria (ALG)        12     5     2     8     15         3   
2          Argentina (ARG)        23    18    24    28     70        18   
3            Armenia (ARM)         5     1     2     9     12         6   
4  Australasia (ANZ) [ANZ]         2     3     4     5     12         0   

   01 !.1  02 !.1  03 !.1  Total.1  ? Games  01 !.2  02 !.2  03 !.2  \
0       0       0       0        0       13       0       0       2   
1       0       0       0        0       15       5       2       8   
2       0       0       0        0       41      18      24      28   
3       0       0       0        0       11       1       2       9   
4       0       0       0        0        2       3       4       5   

   Combined total  
0               2  
1              15  
2              70  
3              12  
4              12

### 1.3 Changer le nom des colonnes en utilisant le dictionnaire suivant :
```python
new_names =  {'Unnamed: 0': 'Country',
              '? Summer': 'Summer Olympics',
              '01 !': 'Gold',
              '02 !': 'Silver',
              '03 !': 'Bronze',
              '? Winter': 'Winter Olympics',
              '01 !.1': 'Gold.1',
              '02 !.1': 'Silver.1',
              '03 !.1': 'Bronze.1',
              '? Games': '# Games', 
              '01 !.2': 'Gold.2',
              '02 !.2': 'Silver.2',
              '03 !.2': 'Bronze.2'}


```



In [35]:
# Changer le nom des colonnes en utilisant le dictionnaire new_names
new_names =  {'Unnamed: 0': 'Country',
              '? Summer': 'Summer Olympics',
              '01 !': 'Gold',
              '02 !': 'Silver',
              '03 !': 'Bronze',
              '? Winter': 'Winter Olympics',
              '01 !.1': 'Gold.1',
              '02 !.1': 'Silver.1',
              '03 !.1': 'Bronze.1',
              '? Games': '# Games', 
              '01 !.2': 'Gold.2',
              '02 !.2': 'Silver.2',
              '03 !.2': 'Bronze.2'}

olympics_df.rename(columns=new_names, inplace=True)







# Afficher les 5 premières lignes

In [36]:
olympics_df.head()

Country  Summer Olympics  Gold  Silver  Bronze  Total  \
0        Afghanistan (AFG)               13     0       0       2      2   
1            Algeria (ALG)               12     5       2       8     15   
2          Argentina (ARG)               23    18      24      28     70   
3            Armenia (ARM)                5     1       2       9     12   
4  Australasia (ANZ) [ANZ]                2     3       4       5     12   

   Winter Olympics  Gold.1  Silver.1  Bronze.1  Total.1  # Games  Gold.2  \
0                0       0         0         0        0       13       0   
1                3       0         0         0        0       15       5   
2               18       0         0         0        0       41      18   
3                6       0         0         0        0       11       1   
4                0       0         0         0        0        2       3   

   Silver.2  Bronze.2  Combined total  
0         0         2               2  
1         2         8              15  
2        24        28              70  
3         2         9              12  
4         4         5              12